br/pkg/stream/stream_mgr.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2015 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stream
+
+import (
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	filter "github.com/pingcap/tidb-tools/pkg/table-filter"
+	"github.com/pingcap/tidb/distsql"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/meta"
+	"github.com/pingcap/tidb/parser/model"
+	"github.com/pingcap/tidb/tablecodec"
+	"github.com/pingcap/tidb/util"
+	"github.com/pingcap/tidb/util/ranger"
+	"go.uber.org/zap"
+)
+
+var metaPrefix = []byte("m")
+
+// appendTableObserveRanges specifies building key ranges [t{tableID}{low}, t{tableID}{high}] corresponding to `tblIDS`
+func appendTableObserveRanges(tblIDs []int64) ([]kv.KeyRange, error) {
+	krs := make([]kv.KeyRange, 0, len(tblIDs))
+	rangers := ranger.FullNotNullRange()
+
+	for _, ran := range rangers {
+		low, high, err := distsql.EncodeIndexKey(nil, ran)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, tid := range tblIDs {
+			startKey := tablecodec.EncodeTablePrefixSeekKey(tid, low)
+			endKey := tablecodec.EncodeTablePrefixSeekKey(tid, high)
```

Comment:
I guess this would be encoded as range `[t{table_id}{0x01}, t{table_id}{0xfa})`, indeed it works, but I'm wandering why not just `[t{table_id}_, NextPrefixKey(t{table_id}_))`... 

(The `Key` type provides [a method](https://pkg.go.dev/github.com/tikv/client-go/v2/kv#PrefixNextKey) generating its prefix next key)